### PR TITLE
Quote all paths in Xcode build backend

### PR DESCRIPTION
**Pull request by cbracken**
Required to handle Flutter SDK (and other) paths that include spaces.

### CODE REVIEW

**Comment by cbracken** [comment]
Part of a series of fixes to #6577

**Comment by cbracken** [comment]
@chinmaygarde @xster

File: packages/flutter_tools/bin/xcode_backend.sh [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 RunCommand() {
   echo "♦ " $@
```
Comment (abarth) [inline packages/flutter_tools/bin/xcode_backend.sh]: Probably should quote this `$@` too, just to be safe.

File: packages/flutter_tools/bin/xcode_backend.sh [hunk 2]
```diff
@@ -54,64 +54,64 @@ BuildApp() {
...
 
-  RunCommand rm -f ${derived_dir}/Flutter.framework
-  RunCommand rm -f ${derived_dir}/app.dylib
-  RunCommand rm -f ${derived_dir}/app.flx
-  RunCommand cp -r ${framework_path}/Flutter.framework ${derived_dir}
-  RunCommand pushd ${project_path}
+  RunCommand rm -f "${derived_dir}/Flutter.framework"
+  RunCommand rm -f "${derived_dir}/app.dylib"
+  RunCommand rm -f "${derived_dir}/app.flx"
+  RunCommand cp -r "${framework_path}/Flutter.framework ${derived_dir}"
```
Comment (abarth) [inline packages/flutter_tools/bin/xcode_backend.sh]: If you really want to be correct, you should add `--` to separate the flags from the file names

**Comment by abarth** [comment]
LGTM

File: packages/flutter_tools/bin/xcode_backend.sh (hunk 1 above)
Comment (cbracken) [inline packages/flutter_tools/bin/xcode_backend.sh]: Done.

File: packages/flutter_tools/bin/xcode_backend.sh (hunk 2 above)
Comment (cbracken) [inline packages/flutter_tools/bin/xcode_backend.sh]: Done.

**Comment by xster** [comment]
LGTM

**Pull request merged**